redbot/core/bot.py

```diff
@@ -406,14 +406,57 @@ async def allowed_by_whitelist_blacklist(
 
         return True
 
-    async def ignored_channel_or_guild(self, ctx: commands.Context) -> bool:
+    async def message_is_eligible_as_command(self, message: discord.Message) -> bool:
+        """
+        Runs through the things which apply globally about commands
+        to determine if a message may be responded to as a command.
+
+        This can't interact with permissions as permissions is hyper-local
+        with respect to command objects, create command objects for this
+        if that's needed.
+
+        This also does not check for prefix or command conflicts,
+        as it is primarily designed for non-prefix based response handling
+        via on_message_without_command
+
+        Parameters
+        ----------
+        message
+            The message object to check
+
+        Returns
+        -------
+        bool
+            Whether or not the message is eligible to be trated as a command.
```

Comment:
```suggestion
            Whether or not the message is eligible to be treated as a command.
```

Reply:
This can't be used to reduce code repetition, unfortunately. This does not interact with permissions, and cannot properly order all the checks to pretend it does.